pkg/machine/e2e/list_test.go

```diff
@@ -182,6 +184,29 @@ var _ = Describe("podman machine list", func() {
 		Expect(listSession).To(Exit(0))
 		Expect(listSession.outputToString()).To(Equal("2GiB 11GiB"))
 	})
+	It("list machine from all providers", func() {
+		skipIfVmtype(define.QemuVirt, "linux only has one provider")
+
+		i := new(initMachine)
+		session, err := mb.setCmd(i.withImage(mb.imagePath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session).To(Exit(0))
+
+		currprovider := os.Getenv("CONTAINERS_MACHINE_PROVIDER")
+		os.Setenv("CONTAINERS_MACHINE_PROVIDER", getOtherProvider())
+
+		i2 := new(initMachine)
+		session2, err := mb.setName("otherprovider").setCmd(i2.withImage(mb.imagePath)).run()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(session2).To(Exit(0))
+		os.Setenv("CONTAINERS_MACHINE_PROVIDER", currprovider)
```

Comment:
The last setenv call must happen as defer before the command otherwise it will never be run if a check above fails then leaking the bad env into all other tests